sql/2024/third-parties/lighthouse_average_unminified_css_by_3p.sql

```diff
@@ -0,0 +1,42 @@
+#standardSQL
+# Pages with unminified JS by 1P/3P
+CREATE TEMPORARY FUNCTION getUnminifiedJsUrls(audit STRING)
+RETURNS ARRAY<STRUCT<url STRING, wastedBytes INT64>> LANGUAGE js AS '''
+try {
+  var $ = JSON.parse(audit);
+  return $.details.items.map(({url, wastedBytes}) => {
+    return {url, wastedBytes};
+  });
+} catch (e) {
+  return [];
+}
+''';
+
+SELECT
+  client,
+  AVG(pct_1p_wasted_bytes) AS avg_pct_1p_wasted_bytes,
+  AVG(pct_3p_wasted_bytes) AS avg_pct_3p_wasted_bytes
+FROM (
+  SELECT
+    client,
+    page,
+    SUM(IF(is_3p, 0, wasted_bytes)) / SUM(wasted_bytes) AS pct_1p_wasted_bytes,
+    SUM(IF(is_3p, wasted_bytes, 0)) / SUM(wasted_bytes) AS pct_3p_wasted_bytes
+  FROM (
+    SELECT
+      _TABLE_SUFFIX AS client,
+      lighthouse.url AS page,
+      NET.HOST(unminified.url) IS NOT NULL AND NET.HOST(unminified.url) IN (
+        SELECT domain FROM `httparchive.almanac.third_parties` WHERE date = '2024-06-01' AND category != 'hosting'
+      ) AS is_3p,
+      unminified.wastedBytes AS wasted_bytes
+    FROM
+      `httparchive.lighthouse.2024_06_01_*` AS lighthouse,
```

Comment:
This is using the old format, right?   This is all in `httparchive.all.pages` now, right?

Reply:
I'm not sure. This is my first year on HTTP Archive. I updated the the dates and it worked for this one. So, I assumed `httparchive.lighthouse.2024_06_01_*` is available for 2024 as well. Please let me know if not and if I should use another table that comprises of all the columns that exist in the query.

Reply:
My first too. @tunetheweb seems to know everything, though.  Is this good as it is, or does this data live in `httparchive.all.pages`?

Reply:
It is available for 2024. But we're trying to encourage everyone to move to the new datamodel now as the Web Almanac are a good source of queries for people so good opportunity to lead by example.

Plus moving to the `httparchive.all.pages` also allows you to query by `is_root_page` to see if there are any differences between home pages and secondary pages for this query, if you wish to. The old table only includes home pages so that's not an option.

Reply:
`httparchive.all.pages` does not have the column `report` as needed by the current query. So, I have kept as it is for now.

Reply:
That’s the `lighthouse` column in that table. It was given a better name than `report` since that generic name isn’t as meaningful in the combined table.

Reply:
Ported this.